chain/actors/aerrors/error.go

```diff
@@ -23,10 +23,17 @@ type internalActorError interface {
 	Unwrap() error
 }
 
+type guard struct{}
+
 type ActorError interface {
 	error
 	IsFatal() bool
 	RetCode() exitcode.ExitCode
+	// guard is a sentinel to prevent code outside if the aerrors package
+	// from implementing the ActorError interface. It ensures actors
+	// use Runtime.Abortf and do not panic with an ActorError directly
+	// and illegally using a system error code.
+	guard() guard
```

Comment:
Just chatted with @Kubuxu. We both agree that this pattern suggests that `ActorError` should really be a struct with an internal `valid bool` field that gets set by the constructors.

But it's not worth doing such a drastic refactor now, hence the interface with a private method that references a private type should be enough.

I'd just recommend calling this method something like `_internal`. The leading `_` is yet another way of keeping methods private (for example `_Internal` would have the same effect), but it's a nice marker to make it evident that this is some form of a type hack.